webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +96,776 @@ public WebSiteSessionController createComponentSessionController(
-          String name = request.getParameter("name");
+    request.setAttribute("SuggestionName", nomSite);
+    request.setAttribute("SuggestionUrl", nomPage);
+    return getDestination("Main", scc, request);
+  }
 
-          /* Supprimer la page */
-          scc.deleteFile(currentPath + "/" + name);
+  private String listSitesMatchingSearch(final WebSiteSessionController scc,
+      final HttpRequest request) {
+    String id = request.getParameter("Id"); /* id de la publication */
 
-          request.setAttribute("Site", scc.getSessionSite());
-        } else if ("classifySite".equals(action)) { // cas de l'upload et du design
-          // CLASSIFY SITE -------------------------------------------------------------
-          // = en cas de new Site ou de classifySite
-          String listeTopics = request.getParameter("ListeTopics");
+    String typeRequest = request.getParameter("Type");
+    String destination = "";
+    if ("Publication".equals(typeRequest)) {
+      // recherche de l'url complete d'acces a la page
+      try {
+        PublicationDetail pubDetail = scc.getPublicationDetail(id);
+        SiteDetail sitedetail = scc.getWebSite(pubDetail.getVersion());
+
+        destination = getWebSitesDestination(sitedetail, request, scc);
+      } catch (Exception e) {
+        SilverLogger.getLogger(this).error(e.getMessage(), e);
+      }
+    } else if ("Site".equals(typeRequest)) {
+      try {
+        SiteDetail sitedetail =  scc.getWebSite(id);
+        destination = getWebSitesDestination(sitedetail, request, scc);
+      } catch (Exception e) {
+        SilverLogger.getLogger(this).error(e.getMessage(), e);
+      }
+    } else if ("Node".equals(typeRequest)) {
+      destination = scc.getComponentUrl() + "listSite.jsp?Action=Search&Id=" + id;
+    }
+    return destination;
+  }
 
-          request.setAttribute("Site", scc.getSessionSite());
+  private String listAllSitesWithinPortlet(final WebSiteSessionController scc,
+      final HttpRequest request, final String userRole) throws WebSitesException {
+    FolderDetail webSitesCurrentFolder = scc.getFolder("0");
+    scc.setSessionTopic(webSitesCurrentFolder);
+    request.setAttribute(CURRENT_FOLDER_PARAM, webSitesCurrentFolder);
 
-          /* publications : classer le site dans les themes cochés */
-          ArrayList<String> arrayToClassify = new ArrayList<>();
-          boolean publish = false;
-
-          ArrayList<String> arrayTopic = new ArrayList<>();
-          int begin = 0;
-          int end = 0;
-          end = listeTopics.indexOf(',', begin);
-          String idTopic = null;
-          while (end != -1) {
-            idTopic = listeTopics.substring(begin, end);
-
-            begin = end + 1;
-            end = listeTopics.indexOf(',', begin);
-
-            arrayTopic.add(idTopic);
-            publish = true;
-          }
-          scc.updateClassification(id, arrayTopic);
+    if (userRole.equals("Publisher") || userRole.equals("Admin")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Admin" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Publisher" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)